### ISSUE TITLE
Incorrect error message when using associated type in const expression

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#44243]
```rust
trait VecN {
    const DIM: usize;
}
trait Mat {
    type Row: VecN;
}

fn m<M: Mat>() {
    let a = [3; M::Row::DIM]; //~ ERROR associated type `Row` not found for `M`
}
fn main() {
}
```
The associated type actually exists. So I started at this a while, wondering whether I just stopped knowing how to read...

(Originally reported at <https://github.com/rust-lang/rust/issues/34344#issuecomment-317902421>.)

**Comment by nwin** [comment]
Duplicate of #44168.

**Comment by shepmaster** [comment]
Closing as duplicate
